### ISSUE TITLE
Errors not thrown

### ISSUE THREAD

**Issue by gonsfx** [jakearchibald/es6-promise#88]
Chrome 41 throws uncatched errors using native `window.Promise`:

```javascript
new window.Promise(function(res, rej) {
  throw new Error('window.Promise');
});
```

The only way i can achieve this behavior using `es6-promise` is by re-throwing in a `setTimeout` in `catch`:

```javascript
var Promise = require('es6-promise');

new Promise(function(res, rej) {
  throw new Error('es6-promise');
}).catch(function(err) {
  setTimeout(function() {
    throw err;
  }, 0);
});
```

This is not developer-friendly at all.
While i've read #70, #71 and especially #40 i don't get how this is specced behavior. Can't find anything in the current ES6 draft on this.

Maybe someone can shed a little light.

**Comment by stefanpenner** [comment]
It actually doesn't throws them, rather an unspeced log. I agree it would be good to add that back to this project.

**Comment by allanhortle** [comment]
Bump. This is strange behaviour.

**Comment by jakearchibald** [comment]
Throwing via setTimeout isn't the right solution here - as Stefan points
out Chrome's doing a simple log, whereas a throw will bubble up to
window.onerror.

**Comment by stefanpenner** [comment]
I've been saturated with other OSS responsibilities, but I will circle around and push this forward :)

**Comment by juandopazo** [comment]
@stefanpenner I can help.

**Comment by stefanpenner** [comment]
> So one way to do this would be:
> 
> On Reject, if there are no error handlers do promise.__errorLogTimeout = setTimeout(() => console.error(error), 5000)
> On then/catch, clearTimeout(promise.__errorLogTimeout).

its has a few more requirements.

I have implemented something (quick and dirty) that closely mimics (but doesn't quite match) the expected heuristics. in RSVP (the parent project) https://github.com/tildeio/rsvp.js/blob/master/lib/rsvp/promise.js#L176-L182  

It would actually be easier to implement it more heuristically correct in this project, as the async isn't pluggable as it is in ember.

So basically, Errors are logged only if not handled before the next macroTask. So their is a concise window where errors need to be handled, and where errors need to be logged.

**Comment by juandopazo** [comment]
Don't you think "the next macrotask" is too soon?

**Comment by stefanpenner** [comment]
> Don't you think "the next macrotask" is too soon?

i don't believe so, we really need the browsers to give us access to their new promise pane....

**Comment by stefanpenner** [comment]
@domenic share your thoughts?

**Comment by domenic** [comment]
I'd just log on next macrotask and log again when it's caught. Chrome still is implementing the last part.

**Comment by stefanpenner** [comment]
duplicate: https://github.com/jakearchibald/es6-promise/issues/70

**Comment by stefanpenner** [comment]
@domenic 
> Chrome still is implementing the last part.

the last part being, giving userland promises access to the new promise pane?

**Comment by domenic** [comment]
No, greying out the log when the rejection gets handled.

**Comment by stefanpenner** [comment]
@domenic lame :P

**Comment by ysmood** [comment]
Event Chrome and Bluebird is not good enough for me. I wrote mine, see the comparison: https://github.com/ysmood/yaku/blob/master/docs/debugHelperComparison.md

For more details: https://github.com/ysmood/yaku
